mousestyles/classification/classification.py

```diff
@@ -0,0 +1,291 @@
+        if not isinstance(n_estimators, list):
+            raise TypeError("n_estimators should be a list")
+        if not all([isinstance(i, int) for i in n_estimators]):
+            raise TypeError("All the n_estimators should be integers")
+        if not all([i > 0 for i in n_estimators]):
+            raise ValueError("All the n_estimators should be positive")
+    if learning_rate is not None:
+        if not isinstance(learning_rate, list):
+            raise TypeError("max_feature should be a list")
+        if not all([i > 0 for i in learning_rate]):
+            raise ValueError("max_features must be greater than 0")
+    # creat GradientBoosting model
+    scaler = StandardScaler()
+    train_x = scaler.fit_transform(train_x)
+    es = n_estimators
+    ls = learning_rate
+    if n_estimators is None:
+        es = [100, 200, 300, 500]
+    if learning_rate is None:
+        ls = list(np.linspace(0.0001, 0.15, 10))
+    gb = GradientBoostingClassifier()
+    clf = GridSearchCV(
+        estimator=gb, param_grid=dict(n_estimators=es, learning_rate=ls),
+        n_jobs=-1)
+    clf.fit(train_x, train_y)
+    clf = clf.best_estimator_
+    # fit the best model
+    clf.fit(train_x, train_y)
+    # predict the testing data and convert to data frame
+    prediction = clf.predict(scaler.fit_transform((test_x)))
+    prediction = pd.DataFrame(prediction)
+    prediction.columns = ['predict_strain']
+    print('The best GradientBoosting Model is:')
+    print(clf)
+    return(prediction)
+
+
+def fit_svm(train_y, train_x, test_x, c=None, gamma=None):
+    """
+    Returns a DataFrame of svm results, containing
+    prediction strain labels and printing the best model. The
+    model's parameters will be tuned by cross validation, and
+    accepts user-defined parameters.
+    Parameters
+    ----------
+    train_y: Series
+             labels of classification results, which are predicted strains.
+    train_x: DataFrame
+             features used to predict strains in training set
+    test_x: DataFrame
+            features used to predict strains in testing set
+    c: list, optional
+       tuning parameter of svm, which is penalty parameter of the error term
+    gamma: list, optional
+           tuning parameter of svm, which is kernel coefficient
+    Returns
+    ----------
+    DataFrame of svm results based on testing strains.
+        Column: prediction strain labels
+    """
+    # input validation
+    if c is not None:
+        if not isinstance(c, list):
+            raise TypeError("c should be a list")
+    if gamma is not None:
+        if not isinstance(gamma, list):
+            raise TypeError("gamma should be a list")
+    # creat svm model
+    scaler = StandardScaler()
+    train_x = scaler.fit_transform(train_x)
+    Cs = c
+    Gammas = gamma
+    if c is None:
+        Cs = list(np.logspace(-6, -1, 10))
+    if gamma is None:
+        Gammas = list(np.linspace(0.0001, 0.15, 10))
+    svc = svm.SVC()
```

Comment:
Probably we might consider adding the kernel choices for SVM for users (or in the future).